### ISSUE TITLE
Correction # 根据物种分布范围扩展裁剪范围

### ISSUE THREAD

**Issue by sayabao** [WeiyeHuang/Tutorial_SDM#2]
# 根据物种分布范围扩展裁剪范围
`extent_crop <- presence_extent + margin * c(-1, 1, -1, 1)`
margin不需要*后面的向量

**Comment by WeiyeHuang** [comment]
Thank you for bringing up this question. In the provided R code, the expression `* c(-1, 1, -1, 1)` is essential for expanding and cropping the extent of the species distribution range. This operation ensures that we add a margin of the specified size (controlled by the 'margin' variable) to the original extent, thereby capturing potential points that might have been missed due to data inaccuracies or noise.

If we omit the `* c(-1, 1, -1, 1)` part and simply use `presence_extent + margin` it would likely result in an error. This is because R requires explicit rules and definitions for operations between different objects. Without the multiplication using `c(-1, 1, -1, 1)`, R wouldn't understand how to combine the extent object with the scalar value (margin).

To ensure the code runs correctly and the species distribution range is appropriately adjusted, it is recommended to retain the `* c(-1, 1, -1, 1)` part in the code.

**Comment by sayabao** [comment]
No, it's wrong. You can set a larger "margin" to test it, for example 10. If we run the code with "* c(-1, 1, -1, 1)", the result is obsolutely not we want.
<img src="https://img.pterclub.com/images/2023/10/10/1111.png" alt="1111.png" border="0" />
